kedro_neptune/datasets.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC
+from typing import Any, Dict
+
+from kedro.io import AbstractVersionedDataSet
+from kedro.io.core import parse_dataset_definition, get_filepath_str
+
+try:
+    # neptune-client=0.9.0+ package structure
+    import neptune.new as neptune
+except ImportError:
+    # neptune-client>=1.0.0 package structure
+    import neptune
+
+
+class AbstractNeptuneDataSet(AbstractVersionedDataSet, ABC):
+    """Abstract class for extending other DataSets.
+     Instances are produced by NeptuneArtifactDataSet factory."""
+    pass
+
+
+class NeptuneArtifactDataSet(AbstractVersionedDataSet):
```

Comment:
AbstractVersionedDataSet works by default on `catalog_neptune.yaml`, but not `neptune_catalog.yaml`.

Reply:
What about simply `AbstractDataSet`? We do we inherits from Versioned one?
Sure, `catalog_neptune.yml` is ok :wink:

Reply:
> What about simply AbstractDataSet?

But `NeptuneArtifactDataSet` was mentioned in spec, right?

> We do we inherits from Versioned one?

Otherwise it's not accepted inside of `kedro`, even if it's simply factory.